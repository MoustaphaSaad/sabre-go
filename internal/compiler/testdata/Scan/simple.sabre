package main

func vertex(pos vec3, normal vec3) (vec4, vec3) {
	gl.Position = projection(pos, 1.0)
	return gl.Position, normal
}

func fragment(color vec3) vec4 {
	return texture(color, 0.5)
}
