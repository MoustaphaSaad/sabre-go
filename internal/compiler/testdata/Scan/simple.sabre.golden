package         "package"               1:1    [0-7]
IDENTIFIER      "main"                  1:9    [8-12]
func            "func"                  3:2    [14-18]
IDENTIFIER      "vertex"                3:7    [19-25]
(               "("                     3:13   [25-26]
IDENTIFIER      "pos"                   3:14   [26-29]
IDENTIFIER      "vec3"                  3:18   [30-34]
,               ","                     3:22   [34-35]
IDENTIFIER      "normal"                3:24   [36-42]
IDENTIFIER      "vec3"                  3:31   [43-47]
)               ")"                     3:35   [47-48]
(               "("                     3:37   [49-50]
IDENTIFIER      "vec4"                  3:38   [50-54]
,               ","                     3:42   [54-55]
IDENTIFIER      "vec3"                  3:44   [56-60]
)               ")"                     3:48   [60-61]
{               "{"                     3:50   [62-63]
IDENTIFIER      "gl"                    4:3    [65-67]
.               "."                     4:5    [67-68]
IDENTIFIER      "Position"              4:6    [68-76]
=               "="                     4:15   [77-78]
IDENTIFIER      "projection"            4:17   [79-89]
(               "("                     4:27   [89-90]
IDENTIFIER      "pos"                   4:28   [90-93]
,               ","                     4:31   [93-94]
LITERAL_FLOAT   "1.0"                   4:33   [95-98]
)               ")"                     4:36   [98-99]
return          "return"                5:3    [101-107]
IDENTIFIER      "gl"                    5:10   [108-110]
.               "."                     5:12   [110-111]
IDENTIFIER      "Position"              5:13   [111-119]
,               ","                     5:21   [119-120]
IDENTIFIER      "normal"                5:23   [121-127]
}               "}"                     6:2    [128-129]
func            "func"                  8:2    [131-135]
IDENTIFIER      "fragment"              8:7    [136-144]
(               "("                     8:15   [144-145]
IDENTIFIER      "color"                 8:16   [145-150]
IDENTIFIER      "vec3"                  8:22   [151-155]
)               ")"                     8:26   [155-156]
IDENTIFIER      "vec4"                  8:28   [157-161]
{               "{"                     8:33   [162-163]
return          "return"                9:3    [165-171]
IDENTIFIER      "texture"               9:10   [172-179]
(               "("                     9:17   [179-180]
IDENTIFIER      "color"                 9:18   [180-185]
,               ","                     9:23   [185-186]
LITERAL_FLOAT   "0.5"                   9:25   [187-190]
)               ")"                     9:28   [190-191]
}               "}"                    10:2    [192-193]
EOF             ""                     11:2    [194-194]
